integration-tests/macos.sh

```diff
@@ -6,8 +6,10 @@ ANY_CA_PEM=integration-tests/one-existing-ca.pem
 ANY_CA_SUBJECT="OU=GlobalSign Root CA - R3, O=GlobalSign, CN=GlobalSign"
 
 reset() {
-  security remove-trusted-cert -d $ANY_CA_PEM || true
-  list | grep "$ANY_CA_SUBJECT"
+  printf "\n*** Remove test CA ***\n"
+  CERT_HASH=$(openssl x509 -in $ANY_CA_PEM -noout -fingerprint -sha1 | cut -d= -f2 | tr -d :)
+  security delete-certificate -Z "$CERT_HASH" /Library/Keychains/System.keychain || true
```

Comment:
Maybe this tidying up is not actually worth it. How about deleting `reset` and all calls to it? I assume GHA gives a fresh, fixed environment for every job run anyway...

Reply:
I assume the goal here was to be able to run it on your local machine, in which case leaving it in the same state it came in would be useful...